core/cache/resident.go

```diff
@@ -229,6 +230,17 @@ func (r *Resident) CacheId() uint64 {
 func (r *Resident) registerWorker(w worker.Worker) func() {
 	id := r.nextResourceId()
 	r.mu.Lock()
+	// If this resident is already being evicted
+	// don't register any new workers.
+	if r.evicting {
+		return func() {
```

Comment:
You're missing an `Unlock` here, as you're holding for the `r.evicting`, but not releasing after the if.

Reply:
Ooops, ty, Fixed.